### ISSUE TITLE
Request to update `log4net` target version (was a question)

### ISSUE THREAD

**Issue by AdamWillden** [aws/aws-logging-dotnet#162]
I have already found the answer below but wanted to post this to help others and form a request that the `log4net` version be updated to at least `2.0.11` if not `2.0.12`

## The Question

I'm facing some confusion, we have installed the latest `log4net` and the latest version of `AWS.Logger.Log4net`. When running the application we are seeing the following log which suggests that something is referencing `log4net` version `2.0.9` not the `2.0.12` we have installed or the `2.0.10` that should be referenced by the `AWS.Logger.Log4net` package.

```
log4net: log4net assembly [log4net, Version=2.0.12.0, Culture=neutral, PublicKeyToken=669e0ddf0bb1aa2a]. Loaded from [C:\Users\adam.willden\AppData\Local\Temp\Temporary ASP.NET Files\vs\6cc2877a\c42c8d35\assembly\dl3\1bcd6fa5\0041b7b0_e2a5d601\log4net.dll]. (.NET Runtime [4.0.30319.42000] on Microsoft Windows NT 10.0.19042.0)
log4net: defaultRepositoryType [log4net.Repository.Hierarchy.Hierarchy]
......
log4net: Created Appender [RollingFileAppender]
log4net: Adding appender named [RollingFileAppender] to logger [Amazon].
log4net: Logger [root] Level string is [ALL].
log4net: Logger [root] level set to [name="ALL",value=-2147483648].
log4net: Loading Appender [AWS] type: [AWS.Logger.Log4net.AWSAppender,AWS.Logger.Log4net]
log4net:ERROR Could not create Appender [AWS] of type [AWS.Logger.Log4net.AWSAppender,AWS.Logger.Log4net]. Reported error follows.
System.IO.FileLoadException: Could not load file or assembly 'log4net, Version=2.0.9.0, Culture=neutral, PublicKeyToken=669e0ddf0bb1aa2a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
File name: 'log4net, Version=2.0.9.0, Culture=neutral, PublicKeyToken=669e0ddf0bb1aa2a'
   at System.RuntimeTypeHandle.GetTypeByName(String name, Boolean throwOnError, Boolean ignoreCase, Boolean reflectionOnly, StackCrawlMarkHandle stackMark, IntPtr pPrivHostBinder, Boolean loadTypeFromPartialName, ObjectHandleOnStack type)
   at System.RuntimeTypeHandle.GetTypeByName(String name, Boolean throwOnError, Boolean ignoreCase, Boolean reflectionOnly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean loadTypeFromPartialName)
   at System.RuntimeType.GetType(String typeName, Boolean throwOnError, Boolean ignoreCase, Boolean reflectionOnly, StackCrawlMark& stackMark)
   at System.Type.GetType(String typeName, Boolean throwOnError, Boolean ignoreCase)
   at log4net.Util.SystemInfo.GetTypeFromString(Assembly relativeAssembly, String typeName, Boolean throwOnError, Boolean ignoreCase)
   at log4net.Util.SystemInfo.GetTypeFromString(String typeName, Boolean throwOnError, Boolean ignoreCase)
   at log4net.Repository.Hierarchy.XmlHierarchyConfigurator.ParseAppender(XmlElement appenderElement)
=== Pre-bind state information ===
LOG: DisplayName = log4net, Version=2.0.9.0, Culture=neutral, PublicKeyToken=669e0ddf0bb1aa2a
 (Fully-specified)
LOG: Appbase = file:///D:/Repos/CAM/cam/source/Dingo.Web/
LOG: Initial PrivatePath = D:\Repos\CAM\cam\source\Dingo.Web\bin
Calling assembly : AWS.Logger.Log4net, Version=3.2.0.0, Culture=neutral, PublicKeyToken=885c28607f98e604.
===
```

Would anyone here have insights into what is going wrong here? Searching for the text `2.0.9` in all files across the codebase yields no results. I can't see a reason on our side why this is happening.

I have just tried to drop the explicit log4net reference and rely on the version pulled in by `AWS.Logger.Log4net`, which should resolve to to `2.0.10` and now see the following showing `2.0.9` which successfully loads:

```
log4net: log4net assembly [log4net, Version=2.0.9.0, Culture=neutral, PublicKeyToken=669e0ddf0bb1aa2a]. Loaded from [C:\Users\adam.willden\AppData\Local\Temp\Temporary ASP.NET Files\vs\6cc2877a\c42c8d35\assembly\dl3\1bcd6fa5\00ffbe6f_7084d601\log4net.dll]. (.NET Runtime [4.0.30319.42000] on Microsoft Windows NT 10.0.19042.0)
```

## The Answer

Found it! https://github.com/apache/logging-log4net/releases/tag/rel%2F2.0.11 
> corrects assembly versioning still reported as 2.0.9 in the 2.0.10 release

So the report of `2.0.9` was incorrect, the assembly is actually `2.0.10`.

`log4net` version `2.0.9` has a CVE which is why upgrading to `2.0.10` was important. But the fact it reports as `2.0.9` makes me uneasy!

Can we look to get a new version out which targets a later `log4net` package without this versioning confusion.

### Environment
* Build Version: 3.2.0
* OS Info: Windows
* Build Environment: Visual Studio 
* Targeted .NET Platform:  ASP.NET 4.8

**Comment by ashishdhingra** [comment]
Hi @AdamWillden,

Thanks for posting your findings. I will discuss this with development team to take it forward.

Thanks,
Ashish

**Comment by ashishdhingra** [comment]
Changed it to `bug` so that it could be prioritized.

**Comment by ashishdhingra** [comment]
[AWS.Logger.Log4net 3.2.1](https://www.nuget.org/packages/AWS.Logger.Log4net/3.2.1) is released which depends on log4net version `2.0.12`.
